p2p/server.py

```diff
@@ -100,7 +100,7 @@ def __init__(self,
         self.max_peers = max_peers
         self.bootstrap_nodes = bootstrap_nodes
         self.preferred_nodes = preferred_nodes
-        if self.preferred_nodes is None and network_id in DEFAULT_PREFERRED_NODES:
+        if not self.preferred_nodes and network_id in DEFAULT_PREFERRED_NODES:
```

Comment:
If we want to specify that there should be no preferred nodes, an empty sequence seems like a reasonable way to do it. So I think the original `is None` check is reasonable here.

Reply:
Indeed, reverting that